### ISSUE TITLE
How custom match with no spaced number and string

### ISSUE THREAD

**Issue by nilooy** [spencermountain/compromise#931]
Love the tool 💯 
i'm having this issue to match string containing number and custom unit with or without space.

i used plugin to add new word to `#Unit` tag 

```js
nlp.plugin({
  words: {
     sotok: 'Unit'
  }
});
```
eg: `#Cardinal #Unit` does match with `24 sotok` but doesn't match with `24sotok`

```js
nlp(`24 sotok`).match("#Cardinal #Unit").out("array") //  ✅ 

nlp(`24sotok`).match("#Cardinal #Unit").out("array") //  ❌
```

I can use regex to do that but is there any way to do with existing tags in compromise?

> There must be something i'm doing wrong as i have seen the exising `money` tag does the same.
```js
const money = nlp("$33.2cad").money().json()
```

**Comment by spencermountain** [comment]
Hey @nilooy this is a very good point. It is a big issue with units, that they are not tokenized properly when the space is dropped. it should be possible to do. See #919 

I can look at this this week, if i have time.

Yeah, a regex should work - if you do something like
```
Let m = doc.match(reg)
m.tag('Value')
m.numbers().json()
```
You'll see that the number parser tries to pick it apart. Im not sure how well it will do, but a proper solution to this is forthcoming.

Cheers!

**Comment by nilooy** [comment]
Perfect, Thanks a lot for the quick response, I will go with regex for now but definitely can't wait to use the proper solution 😄 .

**Comment by nilooy** [comment]
One question on the tag's usage, what is the proper way to capture no spaced words with or without regex
`/^([0-9])/ #Units` - works fine but `/^([0-9])/#Units` doesn't work, is there any existing method to do that ?

**Comment by spencermountain** [comment]
hey, `/^([0-9])/ #Units` will work in next week's release. 
will let you know when it's published
cheers

**Comment by spencermountain** [comment]
hey @nilooy this is fixed now in 14.4.0. 
I wanted to do something that normalized '2l' '2 litre' and '2 litres' but it got too messy.

for now, '2foo' will tokenize as ['Value', 'Unit'], no matter what, and ambiguous units like 'm' will tag asa unit, when following a number
cheers

**Comment by nilooy** [comment]
Hey @spencermountain sorry for late, you are awesome man, thanks a lot for the fix.
Can't wait to try it out.
